### PR TITLE
[#52] Fix CVE-2023-40217.

### LINKED ISSUE #52: Fix CVE-2023-40217.
We might need 3.11.5 https://www.python.org/downloads/release/python-3115/ as this contains an important security fix https://github.com/python/cpython/issues/108310

_Originally posted by @adiroiban in https://github.com/chevah/server/issues/6350#issuecomment-1701944315_

### PR DESCRIPTION
**Pull request by dumol**
Scope
-----

Fixes #52 

Changes
-------

Updates used Python version to 3.11.6 to fix CVE-2023-40217.

**Drive-by changes**:
  * Updates OpenSSL to 3.1.3 for Python's `ssl`  to fix CVE-2023-4807 (where built from sources).
  * Updates `cryptography` to version 41.0.4 with OpenSSL 3.1.3 to fix CVE-2023-4807.
  * Updates SQLite to version 3.43.1 fixing CVE-2023-36191 and CVE-2021-31239.
  * Updates other Python modules to latest version: `setuptools` to 68.2.2, `cffi` to 1.16.0, `setproctitle` to 1.3.3, `charset_normalizer` to 3.3.0.

Testing
-------

Review changes.

Check the automated tests.

### CODE REVIEW

File: pythia.conf [hunk 1]
```diff
@@ -1,11 +1,11 @@
 PYTHON_CONFIGURATION="default@3.11.3.f9d9434"
 # This is defined as a Bash array of options to be passed to commands.
-BASE_REQUIREMENTS=("chevah-brink==1.0.12" "paver==1.2.4")
+BASE_REQUIREMENTS=("chevah-brink==1.0.12" "paver==1.3.4")
```
Comment (adiroiban) [inline pythia.conf:3]:
```suggestion
BASE_REQUIREMENTS=("chevah-brink==1.0.13" "paver==1.3.4")
```

**Review by adiroiban — APPROVED** [review]
I know this is not yet ready to ready.

So far, it looks good.
Many thanks.
Only minor comments

File: pythia.conf (hunk 1 above)
Comment (dumol) [reply pythia.conf]: Thank, commited!

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -94,3 +95,74 @@ jobs:
       with:
         sudo: false
         limit-access-to-actor: true
+
+  linux-arm64:
+    runs-on: laja
```
Comment (adiroiban) [inline .github/workflows/docker.yaml]: I think that is best to start a linux container on macos, and run the action runner inside the container and then have this job run

Comment (dumol) [reply .github/workflows/docker.yaml]: True, the support for launching containers from GitHub's runner is exclusive to Linux.

For later reference, on macOS it errors out with: _Error: Container operations are only supported on Linux runners_

**Comment by dumol** [comment]
The `3.11.6.b5ae34b` testing packages are already used in SFTPPlus 4.33.0.

To avoid their automatic removal, they've been copied manually to `production` at https://bin.chevah.com:20443/production/3.11.6.b5ae34b/ without being released through GitHub releases.

**Pull request merged**